proxy.php

```diff
@@ -117,7 +117,17 @@ protected function build_query() {
 	 */
 	protected function set_search_term() {
 		$search_term = $this->sanitize_string( $_REQUEST['search'] );
-		$this->query = str_replace( '{{ep_placeholder}}', $search_term, $this->query );
+
+		// Stringify the JSON object again just to make the str_replace easier.
+		if ( ! empty( $search_term ) ) {
+			$query_string = json_encode( $this->query );
```

Comment:
Shouldn't we use [wp_json_encode](https://developer.wordpress.org/reference/functions/wp_json_encode/)?

Reply:
This file doesn't run in a WordPress env but is called directly from the user's browser.